pkg/azureutils/logger.go

```diff
@@ -0,0 +1,189 @@
+package azureutils
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2/klogr"
+	diskv1beta1 "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/azuredisk/v1beta1"
+)
+
+type contextKey int
+
+const workflowKey contextKey = 1
+
+type Workflow struct {
```

Comment:
I recommend placing `Workflow` and related functions in a separate `workflow.go` file.